tests/Unit/Numerical/LinearAlgebra/Test_Determinant.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "Numerical/LinearAlgebra/Determinant.hpp"
+
+TEST_CASE("Unit.Numerical.LinearAlgebra.Determinant",
+          "[LinearAlgebra][Numerical][Unit]") {
+  // Catch's Approx has a default tolerance of 100*FLOAT_EPS ~ 10^{-5}.
+  // We want tighter tolerances:
+  Approx approx = Approx::custom().epsilon(1e-15);
+
+  // Test determinant function on general (no symmetry) matrices:
+  // * use rank-2 Tensor in 1-4 dimensions, i.e. 1x1, 2x2, 3x3, 4x4 matrices.
+  // * use Tensor<double, ...>, i.e. data at single spatial point.
+  SECTION("Test general Tensor<double,...> matrices") {
+    // Test determinant of 1d matrix
+    {
+      const tnsr::ij<double, 1, Frame::Grid> matrix(1.3);
+      const auto det = determinant(matrix);
+      CHECK(1.3 == approx(det.get()));
+    }
+
+    // Test determinant of 2d matrix
+    {
+      tnsr::ij<double, 2, Frame::Grid> matrix;
+      matrix.get<0, 0>() = 2.1;
+      matrix.get<1, 0>() = 4.5;
+      matrix.get<0, 1>() = -18.3;
+      matrix.get<1, 1>() = -10.9;
+      const auto det = determinant(matrix);
+      CHECK(59.46 == approx(det.get()));
+    }
+
+    // Test determinant of 3d matrix
+    {
+      tnsr::ij<double, 3, Frame::Grid> matrix;
+      matrix.get<0, 0>() = 1.1;
+      matrix.get<1, 0>() = -10.4;
+      matrix.get<2, 0>() = -4.5;
+      matrix.get<0, 1>() = 3.2;
+      matrix.get<1, 1>() = 15.4;
+      matrix.get<2, 1>() = -19.2;
+      matrix.get<0, 2>() = 4.3;
+      matrix.get<1, 2>() = 16.8;
+      matrix.get<2, 2>() = 2.0;
+      const auto det = determinant(matrix);
+      CHECK(1369.95 == approx(det.get()));
+    }
+
+    // Test determinant of 4d matrix
```

Comment:
I don't really like all these little comments anymore. The information is redundant with the type two lines later and the type can't lie. In general I'm really hesitant on having comments that saying literally exactly what the type says. I think we should just not have these types of comments. (Not your fault, this is a remnant of BitBucket I think)